plugins/view-resources/src/components/filter/ValueFilter.svelte

```diff
@@ -75,47 +84,39 @@
         : {}
     const isDerivedFromSpace = hierarchy.isDerived(_class, core.class.Space)
 
-    const spaces =
-      space !== undefined || isDerivedFromSpace
-        ? []
-        : (
-            await client.findAll(
-              core.class.Space,
-              { archived: { $ne: true } },
-              {
-                projection: {
-                  _id: 1,
-                  archived: 1,
-                  _class: 1
-                }
-              }
-            )
-          ).map((it) => it._id)
-
-    async function doQuery (limit: number | undefined, first1000?: any[]): Promise<boolean> {
+    async function doQuery (limit: number | undefined, sortedValues: any[] | undefined): Promise<boolean> {
       const p = client.findAll(
         _class,
         {
           ...resultQuery,
-          ...(space
+          ...(space !== undefined
             ? { space }
             : isDerivedFromSpace
               ? viewOptions === undefined || viewOptions?.hideArchived === true
                 ? { archived: false }
                 : {}
-              : { space: { $in: spaces } }),
-          ...(first1000 ? { [filter.key.key]: { $nin: first1000 } } : {})
+              : {}),
+          ...(sortedValues !== undefined ? { [prefix + filter.key.key]: { $nin: sortedValues } } : {})
         },
         {
           sort: { modifiedOn: SortingOrder.Descending },
-          projection: { [prefix + filter.key.key]: 1 },
-          ...(limit !== undefined ? { limit } : {})
+          projection: { [prefix + filter.key.key]: 1, space: 1 },
+          ...(limit !== undefined ? { limit } : {}),
+          lookup: {
+            space: core.class.Space
+          }
         }
       )
       if (limit !== undefined) {
         objectsPromise = p
       }
-      const res = await p
+      let res: WithLookup<Doc>[] = await p
+      // We need to filter archived in case it is required
```

Comment:
Maybe better use db for it, and don't lookup space if we don't need it)

Reply:
yes, will do